Classes/Service/AnonymizationService.php

```diff
@@ -99,10 +116,37 @@ protected function getAnonymizedPropertyValuesFor($className)
     public function getAnonymizableClassNames()
     {
         $classNames = $this->reflectionService->getClassNamesByAnnotation(AnonymizableEntity::class);
-        return array_filter($classNames, function($className) {
-            // Only process this className if at least one property should be anonymized
-            return count($this->getAnonymizedPropertyValuesFor($className)) > 0;
-        });
+        return array_filter($classNames, [$this, 'isAnonymizable']);
+    }
+
+    /**
+     * Anonymize a given entity.
+     *
+     * @param object $entity
+     * @param bool $update Whether to update the anonymized entity in it's repository
+     * @return object The anonymized entity
+     * @throws AnonymizationException
+     */
+    public function anonymizeEntity($entity, $update = true)
+    {
+        $className = get_class($entity);
+        if (!$this->isAnonymizable($className)) {
+            throw new AnonymizationException(
+                sprintf('The class %s is not annotated as %s or does not have any anonymizable properties.', $className, AnonymizableEntity::class),
+                1563899393
+            );
+        }
+        $anonymizedPropertyValues = $this->getAnonymizedPropertyValuesFor($className);
+        $this->anonymizeProperties($entity, $anonymizedPropertyValues);
+        if ($update) {
+            try {
+                $repository = $this->getRepositoryFor($className);
+                $repository->update($entity);
+            } catch (\Exception $e) {
+                throw new AnonymizationException('Could not determine a repository for the given entity', 1563899697, $e);
```

Comment:
Please replace the generic exception in the getRepositoryFor with your new AnonymizationException and remove the try/catch here.

Reply:
Repository::update "might" throw an `IllegalObjectTypeException` (which it probably won't because of the way how we got to the repository in the first place), but is more likely to throw an `UnknownObjectException` in case one tries to update a non-persistent single entity.

I will change the type of the exception but will leave this try/catch statement as is.

_Initially_ the AnonyizationException was created to represent errors closer to the anonymization process, so it may be debatable whether there should be a more persistence related exception type in this package, as even the code part highlighted arguably is a bit off my original intention.